### PR TITLE
modify: Add statement timeout query parmeter for MySQL

### PR DESCRIPTION
**Pull request by Dombo**
Similarly to https://github.com/golang-migrate/migrate/pull/327 this adds a statement timeout configuration parameter to the MySQL driver by way of an optional query parameter.

### CODE REVIEW

File: database/mysql/README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 |------------|---------------------|-------------|
 | `x-migrations-table` | `MigrationsTable` | Name of the migrations table |
 | `x-no-lock` | `NoLock` | Set to `true` to skip `GET_LOCK`/`RELEASE_LOCK` statements. Useful for [multi-master MySQL flavors](https://www.percona.com/doc/percona-xtradb-cluster/LATEST/features/pxc-strict-mode.html#explicit-table-locking). Only run migrations from one host when this is enabled. |
+| `x-statement-timeout` | `StatementTimeout` | Abort any statement that takes more than the specified number of milliseconds, functionally similar to [Server-side SELECT statement timeouts](https://dev.mysql.com/blog-archive/server-side-select-statement-timeouts/) but enforced by the client. Available for all versions of MySQL, not just >=5.7. |
```
Comment (Dombo) [inline database/mysql/README.md:9]: In MySQL this parameter is actually referred to as `MAX_EXECUTION_TIME` - it's at the reviewers discretion whether we keep parameters consistent across drivers or try to conform/conflate terminology from different database configuration parameters.

**Comment by Dombo** [comment]
@dhui, @Fontinalis please take a look when you have a chance

Appreciate your efforts!

File: database/mysql/README.md (hunk 1 above)
Comment (dhui) [reply database/mysql/README.md:9]: Thanks for bringing this up. Let's use `x-statement-timeout` for consistency.

File: database/mysql/mysql.go [hunk 2]
```diff
@@ -327,8 +340,15 @@ func (m *Mysql) Run(migration io.Reader) error {
 		return err
 	}
 
+	ctx := context.Background()
+	if m.config.StatementTimeout != 0 {
```
Comment (dhui) [inline database/mysql/mysql.go:344]: I didn't catch this in the postgres PR for this but is there a valid case for negative timeouts? e.g. would anyone want all statements to timeout immediately?

**Review by dhui — COMMENTED** [review]
Thanks for the PR and thanks for linking #327 !

File: database/mysql/mysql.go (hunk 2 above)
Comment (Dombo) [reply database/mysql/mysql.go:344]: Perhaps someone would want to in some testing situation, where they exercise the behaviour of `migrate` in some sort of integration test and want to ensure it times out for a long running migration. For this case you wouldn't need a negative timeout though, an `x-statement-timeout` of say 500ms and a migration containing `DO SLEEP(5);` would suffice in that case.

I can't think of any other situations, can you?

Comment (dhui) [reply database/mysql/mysql.go:344]: > some sort of integration test and want to ensure it times out for a long running migration

Sounds like a valid case to me. Let's continue to allow negative statement timeouts

**Review by dhui — APPROVED** [review]

**Pull request merged**